src/main/java/org/broadinstitute/hellbender/engine/dataflow/datasources/VariantsDataflowSource.java

```diff
@@ -0,0 +1,88 @@
+import com.google.cloud.dataflow.sdk.values.PCollection;
+import htsjdk.tribble.Feature;
+import htsjdk.tribble.FeatureCodec;
+import htsjdk.variant.variantcontext.VariantContext;
+import org.broadinstitute.hellbender.engine.FeatureDataSource;
+import org.broadinstitute.hellbender.engine.FeatureManager;
+import org.broadinstitute.hellbender.exceptions.UserException;
+import org.broadinstitute.hellbender.utils.dataflow.BucketUtils;
+import org.broadinstitute.hellbender.utils.variant.Variant;
+import org.broadinstitute.hellbender.utils.variant.VariantContextVariantAdapter;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+
+/**
+ * VariantsDataflowSource loads variants into PCollections using a local file (GCS Bucket to come).
+ *
+ */
+public class VariantsDataflowSource {
+
+    private final List<String> variantSources;
+    private final Pipeline pipeline;
+
+    /**
+     * VariantsDataflowSource sets up source using files (or eventually GCS buckets).
+     * @param variantSources, list of files (or eventually buckets) to read from
+     * @param pipeline, options to get credentials to access GCS buckets.
+     */
+    public VariantsDataflowSource(final List<String> variantSources, final Pipeline pipeline) {
+        for (final String variantSource : variantSources) {
+            if (BucketUtils.isCloudStorageUrl(variantSource)) {
+                // This problem is tracked with issue 632.
+                throw new UnsupportedOperationException("Cloud storage URIs not supported");
+            }
+        }
+
+        this.variantSources = variantSources;
+        this.pipeline = pipeline;
+    }
+
+    /**
+     * getAllVariants reads variants from file(s) (or eventually GCS bucket(s))
+     * @return a PCollection of variants found in the file.
+     */
+    public PCollection<Variant> getAllVariants() {
+        final List<Variant> aggregatedResults = getVariantsList(variantSources);
+        return pipeline.apply(Create.of(aggregatedResults)).setName("creatingVariants");
+    }
+
+    @SuppressWarnings("unchecked")
+    static FeatureCodec<VariantContext, ?> getCodecForVariantSource( final String variantSource ) {
+        final FeatureCodec<? extends Feature, ?> codec = FeatureManager.getCodecForFile(new File(variantSource));
+        if ( !VariantContext.class.isAssignableFrom(codec.getFeatureType()) ) {
+            throw new UserException(variantSource + " is not in a format that produces VariantContexts");
+        }
+        return (FeatureCodec<VariantContext, ?>)codec;
+    }
+
+
+    static List<Variant> wrapQueryResults( final Iterator<VariantContext> queryResults ) {
+        final List<Variant> wrappedResults = new ArrayList<>();
+        while ( queryResults.hasNext() ) {
+            wrappedResults.add(new VariantContextVariantAdapter(queryResults.next()));
+        }
+        return wrappedResults;
+    }
+
+
+    // For testing.
```

Comment:
This comment is confusing, either `// Visible for testing` or `@VisibleForTesting` would be better.  At first glance I thought this was some sort of helper method for the tests.